### PR TITLE
chore: Move most of the remaining endpoints to zod

### PR DESCRIPTION
**Pull request by sradevski**

### CODE REVIEW

File: packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  )
+
+  if (apiKey.type !== ApiKeyType.PUBLISHABLE) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      "Sales channels can only be associated with publishable API keys"
+    )
+  }
+
+  if (create && create.length) {
```
Comment (fPolic) [inline packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts:34]:
```suggestion
  if ( create?.length) {
```

File: packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      input: workflowInput,
+      throwOnError: false,
+    })
+
+    if (Array.isArray(errors) && errors[0]) {
+      throw errors[0].error
+    }
+  }
+
+  if (toDelete && toDelete.length) {
```
Comment (fPolic) [inline packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts:54]:
```suggestion
  if (toDelete?.length) {
```

**Review by fPolic — APPROVED** [review]
LGTM

File: packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    const workflowInput = {
+      data: [
+        {
+          api_key_id: req.params.id,
+          sales_channel_ids: toDelete,
+        },
+      ],
+    }
+
+    const { errors } = await removeSalesChannelsFromApiKeyWorkflow(
```
Comment (riqwan) [inline packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts:64]: q: any reason why these aren't in one workflow?

Context - if the delete fails, you can't revert the creates. If the case is that only one is running at a time, maybe we can do an XOR validation with zod

**Review by riqwan — APPROVED** [review]
💪🏻

File: packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts (hunk 3 above)
Comment (sradevski) [reply packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts:64]: @riqwan I'll move them to a single workflow, it was just too many changes to do at once so I figured I'll just make the user-facing HTTP interface to what we want it to be first

**Pull request merged**